clients/src/main/java/org/apache/kafka/clients/admin/AbortTransactionResult.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@InterfaceStability.Evolving
+public class AbortTransactionResult {
+    private final Map<TopicPartition, KafkaFutureImpl<Void>> futures;
+
+    AbortTransactionResult(Map<TopicPartition, KafkaFutureImpl<Void>> futures) {
+        this.futures = futures;
+    }
+
+    public KafkaFuture<Void> all() {
```

Comment:
I wonder if we should also add a method to access the result of individual partition. `all()` will fail if any of the partition failed but the user does not know which one and he has no way to inspect the results of all partitions. What do you think?

Reply:
Yeah, that's kind of the typical `all()` contract. We are missing a separate API to get the partition-level results. I can add that. I have also been debating whether to add a placeholder result value just in case we need to return something in the future. What do you think?

Reply:
I realized what I was doing here. Right now, the API is taking only a single `AbortTransactionSpec`, so `all()` has no ambiguity. In the future, we could decide to add batching, but I cannot think of a strong reason for it. It does not make much sense for the hanging transaction cleanup use case. However, if we do decide to, then it will complicate the types in here a bit because the key will probably have to be the `AbortTransactionSpec` itself since `WriteTxnMarkers` does batching both by topic partition and the tuple of `(producerId, producerEpoch, coordinatorEpoch)`. Here I decided to try and keep it simple without committing to a more granular API. Does that seem reasonable?

Reply:
That makes sense. I got confused by the fact that `AbortTransactionResult` takes a `Map` in its constructor. In this case, `all()` seems fine. Thanks for the clarificaiton.